widget/playground/src/containers/StyleLayout/StyleLayout.Preset.tsx

```diff
@@ -0,0 +1,162 @@
+import type { ColorsTypes, PresetTypes } from './StyleLayout.types';
+import type { WidgetColors } from '@rango-dev/widget-embedded';
+
+import { Button, Divider, Switch, Typography } from '@rango-dev/ui';
+import React, { useState } from 'react';
+
+import { PRESETS } from '../../constants';
+import { useConfigStore } from '../../store/config';
+import { shallowEqual } from '../../utils/common';
+
+import {
+  Collapse,
+  Color,
+  ColorsContent,
+  PresetContent,
+  PresetTheme,
+  Row,
+} from './StyleLayout.styles';
+
+const Colors = (props: ColorsTypes) => {
+  return (
+    <ColorsContent>
+      <Color style={{ backgroundColor: props.primary }} />
+      <Color
+        style={{
+          backgroundColor: props.secondary,
+          position: 'absolute',
+        }}
+      />
+      <Color
+        style={{
+          backgroundColor: props.background,
+          position: 'relative',
+        }}
+      />
+    </ColorsContent>
+  );
+};
+
+const HEIGHT_EACH_COL = 41;
+const NUMBER_COL_IN_2_ROW = 4;
+const NUMBER_COL_IN_1_ROW = 2;
+
+export function Preset(props: PresetTypes) {
+  const { tab } = props;
+  const [showMore, setShowMore] = useState({
+    tab,
+    value: false,
+  });
+  const isShowMore = showMore.value && tab === showMore.tab;
+
+  const onChangeTheme = useConfigStore.use.onChangeTheme();
+  const onSelectTheme = useConfigStore.use.onSelectTheme();
+  const { theme } = useConfigStore.use.config();
+  const isDarkTab = tab === 'dark';
+  const isLightTab = tab === 'light';
+  const isAutoTab = tab === 'auto';
+  const PRESETS_FILTER = PRESETS.filter(
+    (preset) =>
+      (isDarkTab && !!preset[tab] && !preset.light) ||
+      (isLightTab && !!preset[tab] && !preset.dark) ||
+      (isAutoTab && !!preset.dark && !!preset.light)
+  );
+  const presetsSize = PRESETS_FILTER.length;
+  const more =
+    presetsSize - (isAutoTab ? NUMBER_COL_IN_1_ROW : NUMBER_COL_IN_2_ROW);
+
+  const height = !isShowMore
+    ? HEIGHT_EACH_COL * 2
+    : HEIGHT_EACH_COL * (isAutoTab ? presetsSize : Math.ceil(presetsSize / 2));
+
+  const onSelectPreset = (customTheme: {
+    dark: WidgetColors;
+    light: WidgetColors;
+  }) => {
+    onSelectTheme(customTheme);
+    onChangeTheme({ name: 'singleTheme', value: isAutoTab ? undefined : true });
+    onChangeTheme({ name: 'mode', value: tab });
```

Comment:
I'm not sure but I don't understand we have called `onChangeTheme` twice here with different values.

Reply:
This function is for changing the items inside the theme object, which changes the corresponding item by getting the name and value